lib/AndroidLayoutFile.js

```diff
@@ -158,13 +158,79 @@ AndroidLayoutFile.prototype.getContext = function() {
     return this.context;
 };
 
-AndroidLayoutFile.prototype.makeKey = function(type, source) {
+AndroidLayoutFile.prototype.makeOldKey = function(type, source) {
     if (type.substring(0, 8) === "android:") {
         type = type.substring(8);
     }
     return (type + '_' + source).replace(/[^a-zA-Z0-9\_]/g, "_");
 };
 
+var reUnicodeChar = /\\u([a-fA-F0-9]{1,4})/g;
+var reOctalChar = /\\([0-8]{1,3})/g;
+
+/**
+ * Unescape the string to make the same string that would be
+ * in memory in the target programming language. This includes
+ * unescaping both special and Unicode characters.
+ *
+ * @static
+ * @param {String} string the string to unescape
+ * @returns {String} the unescaped string
+ */
+AndroidLayoutFile.prototype.unescapeString = function(string) {
+    var unescaped = string;
+
+    while ((match = reUnicodeChar.exec(unescaped))) {
+        if (match && match.length > 1) {
```

Comment:
Are unicode and octal escape sequences even valid in the Android layout xml files?